chord_metadata_service/restapi/middleware.py

```diff
@@ -0,0 +1,57 @@
+from django.conf import settings
+from django.http import HttpResponseForbidden
+import requests
+
+
+# TODO: replace this hacky stuff with more robust solution
+APPLICABLE_ENDPOINTS = frozenset({
+    '/api/individuals',
+    '/api/diseases',
+    '/api/phenotypicfeatures'
+})
+
+
+class CandigAuthzMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        if settings.INSIDE_CANDIG and self.is_applicable_endpoint(request):
+            # TODO: unpack the JWT and get access level
+            access_level = 4
+
+            try:
+                allowed = self.query_opa(request, access_level)
+            except requests.exceptions.RequestException:
+                return HttpResponseForbidden()
+
+            if not allowed:
+                return HttpResponseForbidden()
+
+        response = self.get_response(request)
```

Comment:
What I meant by the early return thing (although it's changed somewhat) is that this could just be
```python
return self.get_response(request)
```
no worries though